prometheus_client/metrics.py

```diff
@@ -203,6 +203,41 @@ def remove(self, *labelvalues: Any) -> None:
             if labelvalues in self._metrics:
                 del self._metrics[labelvalues]
 
+    def remove_matching(self, partial: dict[str, str]) -> int:
```

Comment:
What would you think of `remove_partial_match` for the name of this function? When I first see matching I think that all labelnames/values need to match to be removed not a partial. The name of the variable helps, but could also then be `labels` instead of `partial` which helps with confusion around thinking it could be a partial function.